### ISSUE TITLE
Facebook Callback appends '#_=_' to Return URL

### ISSUE THREAD

**Issue by frenchbread** [jaredhanson/passport#378]
When authenticating with facebook and having this route:
```
router.get('/auth/facebook/callback',
        passport.authenticate('facebook', {
            successRedirect : '/',
            failureRedirect : '/'
        }));
```
.. callback somehow appends `#_=_` to redirect URL.

How can this be fixed?

**Comment by bitsandbytez** [comment]
I know in angular you can set html5 but other than that im not sure.

**Comment by framerate** [comment]
This is coming from facebook's end. More info here:

https://github.com/jaredhanson/passport-facebook/issues/12#issuecomment-61456906

**Comment by frenchbread** [comment]
Related [issue](https://github.com/jaredhanson/passport-facebook/issues/128)

**Comment by jaredhanson** [comment]
Closing.  Not an issue with this project.

**Comment by luisvallejomohl** [comment]
I know this answer is late, but the following code works (at least for me):
**TL;DR:**
```javascript
if(location.hash=='#_=_')location.replace(location.href.split('#')[0]);
```
**More detailed version:**
```javascript
//If #_=_ was appended to the URL:
if(location.hash == '#_=_'){
  //Store the "clean" URL in a variable
  var cleanURL = location.href.split('#')[0];
  //Set the URL to the "clean" URL
  location.replace(cleanURL);
};
```
